docs/mps.md

```diff
@@ -0,0 +1,85 @@
+As processes can not share their CUDA context, running multiple ML model worker on the same GPU results in the creation of multiple CUDA contexts.
+
+For these scenarios NVIDIA offers the Multi-Process Service (MPS) which allows multiple processes to share the same CUDA context and run their kernels in a parallel fashion.
+This can lead to increased performance when using multiple workers on the same GPU as well as a decreased GPU memory utilization due to the shared context.
+
+
+To leverage the benefits of NVIDIA MPS we need to start the MPS daemon with the following commands before starting up TorchServe itself.
+```
+sudo nvidia-smi -c 3
+nvidia-cuda-mps-control -d
+```
+The first command enables the exclusive processing mode for the GPU allowing only one process (the MPS daemon) to utilize it.
+The second command starts the MPS daemon itself.
+To shutdown the daemon we can execute:
+```
+echo quit | nvidia-cuda-mps-control
+```
+For more details on MPS please refer to [NVIDIA's MPS documentation](https://docs.nvidia.com/deploy/mps/index.html).
+It should be noted that MPS only allows 48 processes (for Volta GPUs) to connect to the daemon due limited hardware resources.
+Adding more clients/workers (to the same GPU) will lead to a failure.
+
+## Benchmarks
+To show the performance of TorchServe with activated MPS and help to the decision in enabling MPS for your deployment or not we will perform some benchmarks with representative workloads.
+
+Primarily, we want to investigate how the throughput of a worker evolves with activated MPS for different operation points.
+As an example work load for our benchmark we select the [HuggingFace Transformers Sequence Classification example](https://github.com/pytorch/serve/tree/master/examples/Huggingface_Transformers#sequence-classification).
+We perform the benchmark on a g4dn.4xlarge as well as a p3.2xlarge instance on AWS.
+For the benchmark we concentrate on the model throughput as measured by the [benchmark-ab.py](https://github.com/pytorch/serve/tree/master/benchmarks/benchmark-ab.py) tool.
+
+First, we measure the throughput of a single worker for different batch sizes as it will show us at which point the compute resources of the GPU are fully occupied.
+Second, we measure the throughput with two deployed workers for the batch sizes where we expect the GPUs to have still some resources left over to share.
+For each benchmark we perform five runs and take the median over the runs.
+
+We use the following config.json for the benchmark, only overwriting the number of workers and the batch size accordingly.
+
+```
+{
+    "url":"/home/ubuntu/serve/examples/Huggingface_Transformers/model_store/BERTSeqClassification",
+    "requests": 10000,
+    "concurrency": 600,
+    "input": "/home/ubuntu/serve/examples/Huggingface_Transformers/Seq_classification_artifacts/sample_text_captum_input.txt",
+    "gpus": "1",
```

Comment:
the [gpus](https://github.com/pytorch/serve/blob/master/benchmarks/benchmark-ab.py#L66) of the benchmark-ab.py is only applied on docker container.

Did all of the tests run on docker container?

Reply:
@lxning As discussed, I added the remark regarding the single GPU instance and removed the parameter to avoid confusion.